ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/Conversions.scala

```diff
@@ -246,27 +239,63 @@ private[state] object Conversions {
   /** Encodes a [[BlindingInfo]] into protobuf (i.e., [[DamlTransactionBlindingInfo]]).
     * It is consensus-safe because it does so deterministically.
     */
-  def encodeBlindingInfo(blindingInfo: BlindingInfo): DamlTransactionBlindingInfo =
+  def encodeBlindingInfo(
+      blindingInfo: BlindingInfo,
+      divulgedContracts: Map[ContractId, TransactionOuterClass.ContractInstance],
+  ): DamlTransactionBlindingInfo =
     DamlTransactionBlindingInfo.newBuilder
       .addAllDisclosures(encodeDisclosure(blindingInfo.disclosure).asJava)
-      .addAllDivulgences(encodeDivulgence(blindingInfo.divulgence).asJava)
+      .addAllDivulgences(encodeDivulgence(blindingInfo.divulgence, divulgedContracts).asJava)
       .build
 
-  def decodeBlindingInfo(blindingInfo: DamlTransactionBlindingInfo): BlindingInfo =
+  def decodeBlindingInfo(
+      damlTransactionBlindingInfo: DamlTransactionBlindingInfo
+  ): BlindingInfo = {
+    val blindingInfoDivulgence =
+      damlTransactionBlindingInfo.getDivulgencesList.asScala.iterator.map { divulgenceEntry =>
+        val contractId = decodeContractId(divulgenceEntry.getContractId)
+
+        val divulgedTo = divulgenceEntry.getDivulgedToLocalPartiesList.asScala.toSet
+          .map(Ref.Party.assertFromString)
+
+        contractId -> divulgedTo
+      }.toMap
+
     BlindingInfo(
-      disclosure = blindingInfo.getDisclosuresList.asScala.map { disclosureEntry =>
+      disclosure = damlTransactionBlindingInfo.getDisclosuresList.asScala.map { disclosureEntry =>
```

Comment:
Binding this expression too, e.g. as `bindingInfoDisclosure`, may also improve readability.

Reply:
Done